### PR TITLE
Add caesium blame query package (data-plane-memory-ii W1-γ)

### PR DESCRIPTION
**Pull request by rocketbitz**
## Summary
- New `internal/blame` package: walks a job's append-only `dag_snapshot` rows in commit/time order and attributes each task/edge to the snapshot where its **current descriptor** first appeared (most-recent absent→present transition, not earliest containment).
- **Descriptor-keyed identity** `{name, image, command}`: a same-name task whose image/command changed is a content transition attributed to the **mutating** snapshot; delete-and-readd is attributed to the **re-adding** snapshot.
- **Honestly scoped**: surfaces the introducing `GitCommit` and per-edge `provenance_commit` (commit-only — `DagSnapshot` persists no historical author/ref), and emits an explicit `coverage: "topology+image+command"` caveat because `env`/`spec`/`retries`/`cache`/schema/`sla`/`triggerRules` edits don't write a new snapshot.
- Supports a single-task filter and a `[from, to]` commit range.
- Implements item **C1** (query core). The REST endpoint (C2), CLI (C3), and integration scenario (C4) are follow-on items in Stream C.

## Test plan
- [x] just lint (orchestrator)
- [x] just unit-test (orchestrator) — incl. ordinary introduction, **same-name image-only & command-only mutation**, **delete-and-readd (with survivor + length assertions)**, range-start-after-introduction; all assert the coverage caveat
- [ ] just integration-test — pre-merge gate (C1 is internal; the end-to-end driver lands with C3/C4)
- Opus adversarial review (4 lenses) → 2 confirmed test-coverage fixes applied; 1 medium tiebreak-determinism edge case documented in-code as a deferred substrate enhancement (no monotonic column exists on `dag_snapshot`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/blame/query.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+}
+
+func newTaskKey(task models.DagSnapshotTask) taskKey {
+	command, _ := json.Marshal(task.Command)
+	return taskKey{
+		name:    task.Name,
+		image:   task.Image,
+		command: string(command),
+	}
+}
```
[the remark this answers is not in the document]
Comment (rocketbitz) [reply internal/blame/query.go:209]: Intentional: `json.Marshal` gives an unambiguous, escaping-safe key for the command slice, and `newTaskKey` runs over a bounded set of snapshots per job (read-only diagnostic), so this isn't a hot path. `strings.Join("\\x00")` is marginally cheaper but I'd rather keep the unambiguous encoding. Keeping as-is.

File: internal/blame/query.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+// require an additive monotonic column on dag_snapshot (deferred substrate
+// enhancement, tracked with the broader full-descriptor blame work).
+func sortSnapshotsAscending(snaps []models.DagSnapshot) {
+	sort.Slice(snaps, func(i, j int) bool {
+		if !snaps[i].CreatedAt.Equal(snaps[j].CreatedAt) {
+			return snaps[i].CreatedAt.Before(snaps[j].CreatedAt)
+		}
+		return snaps[i].ID.String() < snaps[j].ID.String()
+	})
+}
```
[the remark this answers is not in the document]
Comment (rocketbitz) [reply internal/blame/query.go:236]: Agreed it allocates; `bytes.Compare(ID[:])` is allocation-free. This tiebreak only fires on an exact `CreatedAt` collision and is documented in the function godoc as a stable-but-arbitrary fallback (dag_snapshot has no monotonic column — an insertion-faithful tiebreak is a tracked substrate follow-up). Since it's an arbitrary order either way over a bounded snapshot set, leaving the string compare for readability alongside that note.

File: internal/blame/query.go [hunk 3]
```diff
@@ -0,0 +1,279 @@
+// Package blame attributes current DAG elements to the snapshot that introduced
+// their persisted topology descriptor.
+package blame
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"sort"
```
[the remark this answers is not in the document]
Comment (rocketbitz) [reply internal/blame/query.go:10]: This pairs with the `strings.Join` / `bytes.Compare` suggestions above; keeping the current `json.Marshal` + `ID.String()` for the reasons noted there, so these imports aren't needed. Thanks.

File: internal/blame/query_test.go [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+}
+
+func task(name, image string, command ...string) models.DagSnapshotTask {
+	return models.DagSnapshotTask{Name: name, Image: image, Command: command}
+}
+
+func edge(from, to, provenanceCommit string) models.DagSnapshotEdge {
+	return models.DagSnapshotEdge{From: from, To: to, ProvenanceCommit: provenanceCommit}
+}
+
```
[the remark this answers is not in the document]
Comment (rocketbitz) [reply internal/blame/query_test.go:243]: Strengthened — `requireTaskIntro` now collects all name+image matches and asserts exactly one (`require.Lenf(..., 1, ...)`), so an ambiguous match fails loudly instead of silently taking the first. (The dedicated command-only test already asserts `Element.Command` directly, but the shared helper is now as strict as the {name,image,command} production key.)

File: internal/blame/query.go [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+		}
+	}
+
+	for _, task := range currentTasks {
+		if opts.Task != "" && task.Name != opts.Task {
+			continue
+		}
+		intro, ok := taskIntroductions[newTaskKey(task)]
+		if !ok {
+			continue
```
[the remark this answers is not in the document]
Comment (rocketbitz) [reply internal/blame/query.go:151]: Fixed — added `if err := ctx.Err(); err != nil { return nil, err }` at the top of the snapshot-walk loop, so a cancelled or deadline-exceeded request stops promptly rather than walking the full history.

File: internal/blame/query.go [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+}
+
+func (q *Query) Blame(ctx context.Context, jobID uuid.UUID, opts Options) (*Result, error) {
+	result := &Result{
+		JobID:      jobID,
+		Coverage:   CoverageTopologyImageCommand,
+		FromCommit: opts.FromCommit,
+		ToCommit:   opts.ToCommit,
+		Tasks:      []TaskAttribution{},
+		Edges:      []EdgeAttribution{},
```
[the remark this answers is not in the document]
Comment (rocketbitz) [reply internal/blame/query.go]: Good distinction. Added a godoc note on `EdgeAttribution.ProvenanceCommit` stating it reflects the *introducing* snapshot's value, and that because `provenance_commit` is excluded from `ContentHash` an edge's provenance can drift without a new snapshot — so it may be stale vs the current descriptor, the same honest limitation as the topology+image+command coverage caveat.
